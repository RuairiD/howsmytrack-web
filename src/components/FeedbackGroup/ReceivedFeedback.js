import React from "react";

import { Card, Empty, List } from "antd";

import FeedbackResponse from "../FeedbackResponse/FeedbackResponse";

const LIST_GRID_LAYOUT = {
    gutter: 16,
    xs: 1,
    sm: 1,
    md: 1,
    lg: 1,
    xl: 1,
    xxl: 2,
};

const ReceivedFeedback = ({ feedbackReceived }) => {
    if (feedbackReceived === null || feedbackReceived === undefined) {
        return (
            <Card>
                <Empty
                    style={{ margin: "1em" }}
                    image={Empty.PRESENTED_IMAGE_DEFAULT}
                    description="Feedback for your submission will appear here once you have submitted feedback for everyone else in your group."
                />
            </Card>
        );
    }

    if (feedbackReceived.length === 0) {
        return (
            <Card>
                <Empty
                    style={{ margin: "1em" }}
                    image={Empty.PRESENTED_IMAGE_SIMPLE}
                    description="Nobody in your group has submitted feedback yet..."
                />
            </Card>
        );
    }

    return (
        <List
            grid={LIST_GRID_LAYOUT}
            dataSource={feedbackReceived}
            renderItem={(feedbackItemReceived) => (
                <List.Item>
                    <FeedbackResponse
                        {...feedbackItemReceived}
                    />
                </List.Item>
            )}
        />
    );
};

export default ReceivedFeedback;
