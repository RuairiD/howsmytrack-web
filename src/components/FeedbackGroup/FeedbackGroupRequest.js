import React from "react";

import { Col, Divider, Row, Typography } from "antd";

import FeedbackRequestSummary from "../FeedbackRequestSummary/FeedbackRequestSummary";
import ReceivedFeedback from "./ReceivedFeedback";

const FeedbackGroupRequest = ({ feedbackRequestSummary, feedbackReceived }) => (
    <React.Fragment>
        <Divider />

        <Row gutter={[16, 16]}>
            <Col>
                <Typography.Title level={3}>Feedback for your submission</Typography.Title>
                <FeedbackRequestSummary
                    feedbackRequestSummary={feedbackRequestSummary}
                />
            </Col>
        </Row>

        <Row gutter={[16, 16]}>
            <Col>
                <ReceivedFeedback feedbackReceived={feedbackReceived} />
            </Col>
        </Row>
    </React.Fragment>
);

export default FeedbackGroupRequest;
