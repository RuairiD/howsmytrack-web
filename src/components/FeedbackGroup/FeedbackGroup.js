import React from "react";

import { Col, List, Result, Row, Typography } from "antd";
import { Div } from "lemon-reset";

import FeedbackResponseForm from "../FeedbackResponseForm/FeedbackResponseForm";
import type { FeedbackResponseFormProps } from "../FeedbackResponseForm/FeedbackResponseForm";

import type { FeedbackResponseProps } from "../FeedbackResponse/FeedbackResponse";

import type { FeedbackRequestSummaryProps } from "../FeedbackRequestSummary/FeedbackRequestSummary";
import FeedbackGroupRequest from "./FeedbackGroupRequest";

const LIST_GRID_LAYOUT = {
    gutter: 16,
    xs: 1,
    sm: 1,
    md: 1,
    lg: 1,
    xl: 1,
    xxl: 2,
};

type Props = {
    feedbackResponseForms: Array<FeedbackResponseFormProps>,
    feedbackReceived: Array<FeedbackResponseProps>,
    feedbackRequestSummary: FeedbackRequestSummaryProps,
};

const FeedbackGroup = ({
    feedbackResponseForms,
    feedbackReceived,
    feedbackRequestSummary,
}: Props) => {
    if (feedbackResponseForms) {
        return (
            <Div>
                <Row gutter={[16, 16]}>
                    <Col>
                        <Typography.Title level={3}>Feedback requests for you</Typography.Title>
                    </Col>
                </Row>
                <List
                    grid={LIST_GRID_LAYOUT}
                    dataSource={feedbackResponseForms}
                    renderItem={(feedbackResponseForm) => (
                        <List.Item>
                            <FeedbackResponseForm
                                {...feedbackResponseForm}
                            />
                        </List.Item>
                    )}
                />

                {feedbackRequestSummary.mediaUrl && (
                    <FeedbackGroupRequest
                        feedbackRequestSummary={feedbackRequestSummary}
                        feedbackReceived={feedbackReceived}
                    />
                )}
            </Div>
        );
    }
    return (
        <Result
            status="error"
            title="You are not authorised to view this group."
        />
    );
};

export default FeedbackGroup;
