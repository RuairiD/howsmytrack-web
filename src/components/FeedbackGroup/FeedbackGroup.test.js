import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FeedbackGroup from "./FeedbackGroup";

jest.mock("react-ga");

describe("FeedbackGroup", () => {
    it("renders an error state if user has no response forms (implies the user isn't a member this group)", () => {
        const wrapper = shallow(
            <FeedbackGroup
                feedbackResponseForms={null}
                feedbackRequestSummary={null}
                feedbackReceived={null}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a List of FeedbackResponseForms for each of user's response forms, but doesn't render a FeedbackGroupRequest for a trackless request", () => {
        const wrapper = shallow(
            <FeedbackGroup
                feedbackResponseForms={[
                    {
                        feedbackResponseId: 1901,
                        currentFeedback: "currentFeedback",
                        mediaUrl: "mediaUrl",
                        mediaType: "mediaType",
                        feedbackPrompt: "feedbackPrompt",
                        alreadySubmitted: true,
                        currentAllowReplies: true,
                        replies: 5,
                        unreadReplies: 4,
                    },
                ]}
                feedbackRequestSummary={{
                    feedbackRequestId: 1901,
                    mediaUrl: null,
                    mediaType: null,
                    feedbackPrompt: null,
                    emailWhenGrouped: false,
                    genre: "genre",
                }}
                feedbackReceived={null}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders a List of FeedbackResponseForms for each of user's response forms and the original request with the track", () => {
        const wrapper = shallow(
            <FeedbackGroup
                feedbackResponseForms={[
                    {
                        feedbackResponseId: 1901,
                        currentFeedback: "currentFeedback",
                        mediaUrl: "mediaUrl",
                        mediaType: "mediaType",
                        feedbackPrompt: "feedbackPrompt",
                        alreadySubmitted: true,
                        currentAllowReplies: true,
                        replies: 5,
                        unreadReplies: 4,
                    },
                ]}
                feedbackRequestSummary={{
                    feedbackRequestId: 1901,
                    mediaUrl: "mediaUrl",
                    mediaType: "mediaType",
                    feedbackPrompt: "feedbackPrompt",
                    emailWhenGrouped: false,
                    genre: "genre",
                }}
                feedbackReceived={[
                    {
                        feedbackResponseId: 1901,
                        feedback: "pretty good",
                        currentRating: 5,
                        allowReplies: false,
                        replies: 3,
                        unreadReplies: 2,
                    },
                ]}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders individual FeedbackResponseForms", () => {
        const wrapper = mount(
            <FeedbackGroup
                feedbackResponseForms={[
                    {
                        feedbackResponseId: 1901,
                        currentFeedback: "currentFeedback",
                        mediaUrl: "mediaUrl",
                        mediaType: "mediaType",
                        feedbackPrompt: "feedbackPrompt",
                        alreadySubmitted: true,
                        currentAllowReplies: true,
                        replies: 5,
                        unreadReplies: 4,
                    },
                ]}
                feedbackRequestSummary={{
                    feedbackRequestId: 1901,
                    mediaUrl: "mediaUrl",
                    mediaType: "mediaType",
                    feedbackPrompt: "feedbackPrompt",
                    emailWhenGrouped: false,
                    genre: "genre",
                }}
                feedbackReceived={[
                    {
                        feedbackResponseId: 1901,
                        feedback: "pretty good",
                        currentRating: 5,
                        allowReplies: false,
                        replies: 3,
                        unreadReplies: 2,
                    },
                ]}
            />,
        );

        expect(wrapper.find("FeedbackResponseForm").length).toBe(1);
    });
});
