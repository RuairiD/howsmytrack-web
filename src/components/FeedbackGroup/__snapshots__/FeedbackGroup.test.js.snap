// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackGroup renders a List of FeedbackResponseForms for each of user's response forms and the original request with the track 1`] = `
<Div
  className=""
>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <Title
        level={3}
      >
        Feedback requests for you
      </Title>
    </Col>
  </Row>
  <List
    bordered={false}
    dataSource={
      Array [
        Object {
          "alreadySubmitted": true,
          "currentAllowReplies": true,
          "currentFeedback": "currentFeedback",
          "feedbackPrompt": "feedbackPrompt",
          "feedbackResponseId": 1901,
          "mediaType": "mediaType",
          "mediaUrl": "mediaUrl",
          "replies": 5,
          "unreadReplies": 4,
        },
      ]
    }
    grid={
      Object {
        "gutter": 16,
        "lg": 1,
        "md": 1,
        "sm": 1,
        "xl": 1,
        "xs": 1,
        "xxl": 2,
      }
    }
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
  />
  <FeedbackGroupRequest
    feedbackReceived={
      Array [
        Object {
          "allowReplies": false,
          "currentRating": 5,
          "feedback": "pretty good",
          "feedbackResponseId": 1901,
          "replies": 3,
          "unreadReplies": 2,
        },
      ]
    }
    feedbackRequestSummary={
      Object {
        "emailWhenGrouped": false,
        "feedbackPrompt": "feedbackPrompt",
        "feedbackRequestId": 1901,
        "genre": "genre",
        "mediaType": "mediaType",
        "mediaUrl": "mediaUrl",
      }
    }
  />
</Div>
`;

exports[`FeedbackGroup renders a List of FeedbackResponseForms for each of user's response forms, but doesn't render a FeedbackGroupRequest for a trackless request 1`] = `
<Div
  className=""
>
  <Row
    gutter={
      Array [
        16,
        16,
      ]
    }
  >
    <Col>
      <Title
        level={3}
      >
        Feedback requests for you
      </Title>
    </Col>
  </Row>
  <List
    bordered={false}
    dataSource={
      Array [
        Object {
          "alreadySubmitted": true,
          "currentAllowReplies": true,
          "currentFeedback": "currentFeedback",
          "feedbackPrompt": "feedbackPrompt",
          "feedbackResponseId": 1901,
          "mediaType": "mediaType",
          "mediaUrl": "mediaUrl",
          "replies": 5,
          "unreadReplies": 4,
        },
      ]
    }
    grid={
      Object {
        "gutter": 16,
        "lg": 1,
        "md": 1,
        "sm": 1,
        "xl": 1,
        "xs": 1,
        "xxl": 2,
      }
    }
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
  />
</Div>
`;

exports[`FeedbackGroup renders an error state if user has no response forms (implies the user isn't a member this group) 1`] = `
<Result
  status="error"
  title="You are not authorised to view this group."
/>
`;
